### PR TITLE
autox chart

### PR DESCRIPTION
**Pull request by sriumcp**
Signed-off-by: Srinivasan Parthasarathy <spartha@us.ibm.com>

This is step 1 towards addressing iter8-tools/iter8#1252

### CODE REVIEW

File: charts/autox/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+**Is your feature request related to a problem? Please describe.**
```
Comment (kalantar) [inline charts/autox/README.md:1]: This is the issue text; is that what we want here?

File: charts/autox/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+--set groups.<group name>.chart=<name of iter8 experiment chart> \
+--set groups.<group name>.resourceKind=deployment \
+--set groups.<group name>.namespaces={a,list,of,namespaces} \
+--set groups.<group name>.values.x=y \
+--set groups.<group name>.values.x=y \
+...
+```
+In the above syntax, `x=y` are the parameter values you would set when launching the experiment using `iter8 k launch`.
+
+Following is an **example** of how the above  solution can be used. Set up autox to kick start a `load-test-http` experiment whenever there are changes deployments with an iter8-specific group annotation.
```
Comment (Alan-Cha) [inline charts/autox/README.md:18]:
```suggestion
The following is an **example** of how the above solution can be used. Set up autox to kick start a `load-test-http` experiment whenever there are changes deployments with an iter8-specific group annotation.
```

File: charts/autox/README.md (hunk 1 above)
Comment (sriumcp) [reply charts/autox/README.md]: Fixed.

File: charts/autox/README.md (hunk 2 above)
Comment (sriumcp) [reply charts/autox/README.md]: Fixed.

File: charts/autox/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+kubectl annotate deploy/nginx iter8.tools/group=hello
+```
+
+A few seconds later ... you can view the report for the auto launched experiment using familiar `iter8` commands.
+```shell
+iter8 k report -g hello
+```
+
+
+Should automatically trigger an `http-load-test` for the `new-sevice`.
```
[the remark this answers is not in the document]
Comment (sriumcp) [reply charts/autox/README.md]: Fixed.

**Review by kalantar — APPROVED** [review]
lgtm

File: charts/autox/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    matchLabels:
+      app.kubernetes.io/name: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ .Release.Name }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+    spec:
+      serviceAccountName: {{ .Release.Name }}
```
Comment (kalantar) [inline charts/autox/templates/deployment.yaml:17]: `serviceAccountName` should be a child of the container

**Review by kalantar — CHANGES_REQUESTED** [review]

File: charts/autox/templates/deployment.yaml (hunk 4 above)
Comment (sriumcp) [reply charts/autox/templates/deployment.yaml:17]: I don't think so. `serviceAccountName` belongs to the pod template spec as described above. Perhaps [this accepted answer helps](https://stackoverflow.com/questions/44505461/how-to-configure-a-non-default-serviceaccount-on-a-deployment)?

Comment (kalantar) [reply charts/autox/templates/deployment.yaml:17]: You are right. My mistake. Misread the reference doc.

**Review by kalantar — APPROVED** [review]
lgtm

**Pull request merged**